core/runtime.rs

```diff
@@ -349,6 +357,36 @@ impl JsRuntime {
       .unwrap();
   }
 
+  // Inits JS of provided JsRuntimeModules
+  // NOTE: this will probably change when streamlining snapshot flow
+  pub fn init_mod_js(&mut self) -> Result<(), AnyError> {
+    let mut modules: Vec<Box<dyn JsRuntimeModule>> =
+      self.modules.drain(..).collect();
+    let selfish = Rc::new(RefCell::new(self));
+    for m in modules.iter_mut() {
+      let js_files = m.init_js()?;
+      for (filename, source) in js_files {
+        selfish.borrow_mut().execute_static(filename, source)?;
+      }
+    }
+    selfish.borrow_mut().modules = modules;
```

Comment:
@AaronO could you explain the reason for "taking" modules, wrapping self in `Rc<RefCell<>>` and then putting modules back?

Reply:
To avoid double-borrows, I've simplified it to remove the self wrap.